pkg/metrics/metrics.go

```diff
@@ -647,6 +724,13 @@ type LegacyMetrics struct {
 	APILimiterRateLimit              metric.Vec[metric.Gauge]
 	APILimiterAdjustmentFactor       metric.Vec[metric.Gauge]
 	APILimiterProcessedRequests      metric.Vec[metric.Counter]
+	WorkQueueDepth                   metric.Vec[metric.Gauge]
```

Comment:
Might be clearer to encapsulate these into their own type aka `workQueueMetrics` and aggregate into LegacyMetrics?

Reply:
Ah... I stand corrected. Good catch! Thank you Yusuke!! Please disregard.